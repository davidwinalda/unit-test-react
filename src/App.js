import React, { useEffect, useState } from 'react';
import axios from 'axios';

import './App.css';
import Account from './components/Account';

const App = () => {
  const [users, setUsers] = useState();
  const [isLoading, setIsLoading] = useState(true);
  const [error, setError] = useState();

  // Fecth data API
  const fetchUser = () => {
    axios.get('https://jsonplaceholder.typicode.com/users')
    .then((response) => {
      console.log(response.data)
      setUsers(response.data);
      setIsLoading(false);
    })
    .catch((error) => {
      setError(error)
    })
  }

  useEffect(() => {
    fetchUser()
  }, [])

  return (
    <div className="App">
      <h1>Display Active Users Account Details</h1>
      {
        isLoading ? (
          <p>Fetching users...</p>
        ) : (
          users.map((user) => {
            return <Account key={user.username} user={user} />;
          })
        )
      }
    </div>
  );
}

export default App;
