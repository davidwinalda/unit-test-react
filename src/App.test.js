import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from "enzyme-to-json";

import App from './App';
import Account from './components/Account';

it('Pages App running without crash', () => {
  shallow(<App />);
});

it('Render account header', () => {
  const wrapper = shallow(<App />);
  const header = <h1>Display Active Users Account Details</h1>
  expect(wrapper.contains(header)).toEqual(true);
})

const user = {
  name: 'David Winalda',
  email: 'davidwinalda94@gmail.com',
  username: 'davidwinalda',
}

describe('Test Account Component', () => {
  it('accept user account props', () => {
    const wrapper = mount(<Account user={user} />);
    expect(wrapper.props().user).toEqual(user);
  });

  it('contains users account email', () => {
    const wrapper = mount(<Account user={user} />);
    const email = wrapper.find("p").text();
    expect(email).toEqual("davidwinalda94@gmail.com");
  });
})

it("renders correctly", () => {
  const tree = mount(<Account user={user} />);
  expect(toJson(tree)).toMatchSnapshot();
});

