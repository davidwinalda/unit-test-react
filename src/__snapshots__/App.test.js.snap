// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Account
  user={
    Object {
      "email": "davidwinalda94@gmail.com",
      "name": "David Winalda",
      "username": "davidwinalda",
    }
  }
>
  <div
    key="davidwinalda"
    style={
      Object {
        "margin": "2rem 0rem 0rem 15rem",
        "width": "15rem",
      }
    }
  >
    <h4>
      David Winalda
    </h4>
    <p>
      davidwinalda94@gmail.com
    </p>
    <hr />
  </div>
</Account>
`;
